api/src/main/java/org/openmrs/module/legacyui/api/impl/LegacyUIImpl.java

```diff
@@ -0,0 +1,291 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.legacyui.api.impl;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Concept;
+import org.openmrs.Location;
+import org.openmrs.Obs;
+import org.openmrs.Order;
+import org.openmrs.Patient;
+import org.openmrs.PatientProgram;
+import org.openmrs.PatientState;
+import org.openmrs.ProgramWorkflow;
+import org.openmrs.ProgramWorkflowState;
+import org.openmrs.Provider;
+import org.openmrs.User;
+import org.openmrs.api.APIException;
+import org.openmrs.api.OrderService;
+import org.openmrs.api.ProviderService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.impl.BaseOpenmrsService;
+import org.openmrs.module.legacyui.api.LegacyUIService;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Default implementation of the patient service. This class should not be used on its own. The
+ * current OpenMRS implementation should be fetched from the Context via
+ * <code>Context.getPatientService()</code>
+ * 
+ * @see org.openmrs.api.context.Context
+ * @see org.openmrs.api.PatientService
+ * @see org.openmrs.api.PersonService
+ */
+@Transactional
+public class LegacyUIImpl extends BaseOpenmrsService implements LegacyUIService {
+	
+	private final Log log = LogFactory.getLog(this.getClass());
+	
+	/**
+	 * Copied from OpenMRS core 1.12.x and added back #discontinueAllDrugOrders from 1.9.x
+	 * See https://github.com/openmrs/openmrs-core/blob/1.12.x/api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java#L1191
```

Comment:
A line from 1.9.x or 1.12.x may point to something very different, a few years from now. You need to use a tag.

Reply:
@dkayiwa I'm not sure what to use here. Do you mean something like:
@see org.openmrs.api.VisitService#getAllVisitTypes()
How can we specificaly point to 1.9.x and 1.12.x methods ?

Reply:
e.g: https://github.com/openmrs/openmrs-core/blob/1.12.1/api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java#L1191

Reply:
Got it. I made a fix [here](https://github.com/openmrs/openmrs-module-legacyui/pull/128/commits/9dd92c8a5560377daedb2e231c639e315fa177b8).